Sources/TensorFlow/Operators/NN.swift

```diff
@@ -293,279 +406,148 @@ public extension Tensor where Scalar: TensorFlowFloatingPoint {
-        where Scalar : TensorFlowFloatingPoint
-    )
-    func averagePooled2D(
-        kernelSize: (Int, Int, Int, Int),
-        strides: (Int, Int, Int, Int),
-        padding: Padding
-    ) -> Tensor {
-        return Raw.avgPool(
-            value: self,
-            ksize: [Int32(kernelSize.0), Int32(kernelSize.1),
-                    Int32(kernelSize.2), Int32(kernelSize.3)],
-            strides: [Int32(strides.0), Int32(strides.1), Int32(strides.2), Int32(strides.3)],
-            padding: padding.raw)
-    }
+/// Computes a 2-D average pooling, with the specified filter sizes, strides,
+/// and padding.
+///
+/// - Parameters:
+///     - filterSize: The dimensions of the pooling kernel.
+///     - strides: The strides of the sliding filter for each dimension of the
+///         input.
+///     - padding: The padding for the operation.
+@differentiable(wrt: x, vjp: _vjpAvgPool2D(filterSize:strides:padding:))
+func avgPool2D<Scalar: TensorFlowFloatingPoint>(
+    filterSize: (Int, Int, Int, Int),
+    strides: (Int, Int, Int, Int),
+    padding: Padding
+    ) -> Tensor<Scalar> {
+    return Raw.avgPool(
+        value: x,
+        ksize: [Int32(filterSize.0), Int32(filterSize.1),
+                Int32(filterSize.2), Int32(filterSize.3)],
+        strides: [Int32(strides.0), Int32(strides.1), Int32(strides.2), Int32(strides.3)],
+        padding: padding.raw)
+}
 
-    /// Computes a 3-D average pooling, with the specified kernel sizes, strides,
-    /// and padding.
-    ///
-    /// - Parameters:
-    ///     - kernelSize: The dimensions of the pooling kernel.
-    ///     - strides: The strides of the sliding filter for each dimension of the
-    ///         input.
-    ///     - padding: The padding for the operation.
-    @inlinable @inline(__always)
-    @differentiable(
-        wrt: self, vjp: _vjpAveragePooled3D(kernelSize:strides:padding:)
-        where Scalar : TensorFlowFloatingPoint
-    )
-    func averagePooled3D(
-        kernelSize: (Int, Int, Int, Int, Int),
-        strides: (Int, Int, Int, Int, Int),
-        padding: Padding
-    ) -> Tensor {
-        return Raw.avgPool3D(
-	    self,
-            ksize: [Int32(kernelSize.0), Int32(kernelSize.1),
-                    Int32(kernelSize.2), Int32(kernelSize.3), Int32(kernelSize.4)],
-            strides: [Int32(strides.0), Int32(strides.1), Int32(strides.2), Int32(strides.3),
-                      Int32(strides.4)],
-            padding: padding.raw)
-    }
+/// Computes a 3-D average pooling, with the specified filter sizes, strides,
+/// and padding.
+///
+/// - Parameters:
+///     - filterSize: The dimensions of the pooling kernel.
+///     - strides: The strides of the sliding filter for each dimension of the
+///         input.
+///     - padding: The padding for the operation.
+@differentiable(wrt: x, vjp: _vjpAvgPool3D(filterSize:strides:padding:))
+func avgPool3D<Scalar : TensorFlowFloatingPoint>(
```

Comment:
Leave no space before a colon in a generic parameter list.

```suggestion
func avgPool3D<Scalar: TensorFlowFloatingPoint>(
```